Problem with `ScatterMax` and `AmrTag` enums

@bprather came across this during the developer meeting with some combination of kokkos/backends where the `ScatterMax` reduction doesn't know what to do with the `AmrTag` enum. If I remember correctly a quick fix was to cast to an int for the reduction, and cast back to `AmrTag`s after.

Possibly related to the Cuda/OpenMP kokkos backend combination from Phillip Edelman:

```
    292    /tmp/pedelmann/spack-stage/spack-stage-parthenon-25.05-7kyih2b6xi2vtcjbzh4xqsqym4qcekhc/spack-src/src/parthenon_array_generic.hpp:226:34:   required from 'aut
            o parthenon::ParArrayGeneric<Data, State, <template-parameter-1-3> >::ToScatterView() [with Op = Kokkos::Experimental::ScatterMax; Data = Kokkos::View<parthen
            on::AmrTag*, Kokkos::LayoutRight, Kokkos::HostSpace>; State = parthenon::empty_state_t; <template-parameter-1-3> = int]'
     293      226 |     return Kokkos::Experimental::ScatterView<data_type, layout, exec_space, Op>(data_);
     294          |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     295    /tmp/pedelmann/spack-stage/spack-stage-parthenon-25.05-7kyih2b6xi2vtcjbzh4xqsqym4qcekhc/spack-src/src/amr_criteria/refinement_package.cpp:115:79:   required f
            rom here
     296      115 |   auto scatter_tags = amr_tags.ToScatterView<Kokkos::Experimental::ScatterMax>();
     297          |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
  >> 298    /home/pedelmann/local/spack/opt/spack/linux-skylake/kokkos-4.6.01-asyrki2oxe22qivmdqbp33qopd75blzl/include/Kokkos_ScatterView.hpp:430:47: error: incomplete ty
            pe 'Kokkos::reduction_identity<parthenon::AmrTag>' used in nested name specifier
     299      430 |     value = reduction_identity<ValueType>::max();
     300          |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
     301    [ 96%] Building CXX object src/CMakeFiles/parthenon.dir/globals.cpp.o
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,7 @@
 - [[PR 1253]](https://github.com/parthenon-hpc-lab/parthenon/pull/1253) Add support for uint64 swarm variables and add default id
 
 ### Fixed (not changing behavior/API/variables/...)
+- [[PR 1276]](https://github.com/parthenon-hpc-lab/parthenon/pull/1276) Specialize Kokkos::reduction_identity for AmrTag
 - [[PR 1275]](https://github.com/parthenon-hpc-lab/parthenon/pull/1275) Remove typeid from interpolation device code
 - [[PR 1254]](https://github.com/parthenon-hpc-lab/parthenon/pull/1254) Fix task failure handling
 - [[PR 1272]](https://github.com/parthenon-hpc-lab/parthenon/pull/1272) Remove mistakenly included pdf files in the base directory

--- a/src/amr_criteria/refinement_package.hpp
+++ b/src/amr_criteria/refinement_package.hpp
@@ -51,7 +51,5 @@ void SecondDerivative(const AMRBounds &bnds, MeshData<Real> *md, const std::stri
                       const int max_level_);
 
 } // namespace Refinement
-
 } // namespace parthenon
-
 #endif // AMR_CRITERIA_REFINEMENT_PACKAGE_HPP_

--- a/src/basic_types.hpp
+++ b/src/basic_types.hpp
@@ -251,4 +251,15 @@ template <typename T>
 using Triple_t = std::tuple<T, T, T>;
 } // namespace parthenon
 
+namespace Kokkos {
+// this specialization is needed for AmrTags to be used as the type in a
+// Kokkos::ScatterView
+template <>
+struct reduction_identity<parthenon::AmrTag> {
+  using AmrTag = parthenon::AmrTag;
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static AmrTag max() { return AmrTag::derefine; }
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static AmrTag min() { return AmrTag::refine; }
+};
+} // namespace Kokkos
+
 #endif // BASIC_TYPES_HPP_

